examples/idempotency/src/working_with_idempotent_function_pydantic_output_serializer.py

```diff
@@ -0,0 +1,44 @@
+from aws_lambda_powertools.utilities.idempotency import (
+    DynamoDBPersistenceLayer,
+    IdempotencyConfig,
+    idempotent_function,
+)
+from aws_lambda_powertools.utilities.idempotency.serialization.pydantic import PydanticSerializer
+from aws_lambda_powertools.utilities.parser import BaseModel
+from aws_lambda_powertools.utilities.typing import LambdaContext
+
+dynamodb = DynamoDBPersistenceLayer(table_name="IdempotencyTable")
+config = IdempotencyConfig(event_key_jmespath="order_id")  # see Choosing a payload subset section
+
+
+class OrderItem(BaseModel):
+    sku: str
+    description: str
+
+
+class Order(BaseModel):
+    item: OrderItem
+    order_id: int
+
+
+class OrderOutput(BaseModel):
+    order_id: int
+
+
+@idempotent_function(
+    data_keyword_argument="order",
+    config=config,
+    persistence_store=dynamodb,
+    output_serializer=PydanticSerializer(model=OrderOutput),
+)
+def process_order(order: Order) -> OrderOutput:
```

Comment:
This sounds like a nice idea, I'll investigate and check

Reply:
Hey @rubenfonseca, I've added this behavior and developed this flow in a separate branch (in order to keep things clean)
please take a look here
https://github.com/aradyaron/powertools-lambda-python/pull/2 and let me know your opinion.

if you prefer that I'll merge and then we'll discuss please let me know

Reply:
I've checked the branch and I think this will be a much better experience. I also liked that you made it generic enough to support other data types! Please merge the code, and we can review it here (just small naming things here and there). Awesome!